geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java

```diff
@@ -9665,7 +9665,9 @@ public String toString() {
 
     @Override
     public boolean equals(Object other) {
-      // TODO: equals should always check the class of other
+      if (!(other instanceof IndexTask)) {
+        return false;
+      }
       if (other == null) {
```

Comment:
same, this can probably be removed since the instanceof check will return early if other is null

Reply:
my bad, I missed that!!